node/internal/child_process.ts

```diff
@@ -470,3 +474,100 @@ function buildCommand(
   }
   return [file, args];
 }
+
+function _createSpawnSyncError(
+  status: string,
+  command: string,
+  args: string[] = [],
+): ErrnoException {
+  const error = errnoException(
+    codeMap.get(status),
+    "spawnSync " + command,
+  );
+  error.path = command;
+  error.spawnargs = args;
+  return error;
+}
+
+export interface SpawnSyncOptions {
+  cwd?: string | URL;
+  input?: string | Buffer | DataView;
+  argv0?: string;
+  stdio?: Array<NodeStdio | number | null | undefined | Stream> | NodeStdio;
+  env?: Record<string, string>;
+  uid?: number;
+  gid?: number;
+  timeout?: number;
+  maxBuffer?: number;
+  encoding?: string;
+  shell?: boolean | string;
+  windowsVerbatimArguments?: boolean;
+  windowsHide?: boolean;
+}
+
+export interface SpawnSyncResult {
+  pid?: number;
+  output?: [string | null, string | Buffer, string | Buffer];
+  stdout?: Buffer | string;
+  stderr?: Buffer | string;
+  status?: number | null;
+  signal?: string | null;
+  error?: Error;
+}
+
+export function spawnSync(
+  command: string,
+  args: string[],
+  options: SpawnSyncOptions,
+): SpawnSyncResult {
+  const {
+    env = Deno.env.toObject(),
+    stdio = ["pipe", "pipe", "pipe"],
+    shell = false,
+    cwd,
+    encoding,
+    uid,
+    gid,
+    maxBuffer,
+  } = options;
+  const normalisezedStdio = normalizeStdioOption(stdio);
+  [command, args] = buildCommand(command, args ?? [], shell);
+
+  const result: SpawnSyncResult = {};
+  try {
+    const output = Deno.spawnSync(command, {
+      args,
+      cwd,
+      env,
+      stdout: toDenoStdio(normalisezedStdio[1] as NodeStdio | number),
+      stderr: toDenoStdio(normalisezedStdio[2] as NodeStdio | number),
+      uid,
+      gid,
+    });
+
+    const { signal } = output;
+    const status = signal ? null : 0;
+    let stdout = Buffer.from(output.stdout) as string | Buffer;
+    let stderr = Buffer.from(output.stderr) as string | Buffer;
+
+    if (stdout.length > maxBuffer! || stderr.length > maxBuffer!) {
+      result.error = _createSpawnSyncError("ENOBUFS", command, args);
+    }
+
+    if (encoding && encoding !== "buffer") {
+      stdout = stdout.toString(encoding);
+      stderr = stderr.toString(encoding);
+    }
```

Comment:
Is it possible to avoid converting to a Buffer and then potentially converting to a string? It would be nice if we only needed to do one conversion.